sdk/search/azure-search/src/main/java/com/azure/search/SearchPagedResponse.java

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.rest.Page;
+import com.azure.core.http.rest.PagedResponseBase;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.CoreUtils;
+import com.azure.search.models.FacetResult;
+import com.azure.search.models.SearchDocumentsResult;
+import com.azure.search.models.SearchRequest;
+import com.azure.search.models.SearchResult;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+/**
+ * Represents an HTTP response from the search API request
+ * that contains a list of items deserialized into a {@link Page}.
+ * Each page contains additional information returned by the API request. In the Search API case
+ * the additional information is:
+ * count - number of total documents returned. Will be returned only if isIncludeTotalResultCount is set to true
+ * coverage - coverage value.
+ */
+public class SearchPagedResponse extends PagedResponseBase<String, SearchResult> {
+
+    private final Map<String, List<FacetResult>> facets;
+    private final Long count;
+    private final Double coverage;
+
+    /**
+     * Constructor
+     *
+     * @param documentSearchResponse an http response with the results
+     */
+    public SearchPagedResponse(SimpleResponse<SearchDocumentsResult> documentSearchResponse) {
+        super(documentSearchResponse.getRequest(),
+            documentSearchResponse.getStatusCode(),
+            documentSearchResponse.getHeaders(),
+            documentSearchResponse.getValue().getResults(),
+            CoreUtils.isNullOrEmpty(documentSearchResponse.getValue().getNextLink())
+                || documentSearchResponse.getValue().getNextPageParameters() == null
+                || documentSearchResponse.getValue().getNextPageParameters().getSkip() == null
```

Comment:
@navalev Why this condition on skip? If there is a `nextPageParameters` annotation in the response, that means there is another page to fetch, and skip will be set accordingly. Did you encounter a situation where it wasn't?

Reply:
@brjohnstmsft this is validation of values before passing them on.

Reply:
@navalev Is it necessary though? `serialize` just converts the result of `getNextPageParameters` to a string. That should tolerate a null value for `skip`, shouldn't it? This looks like a bug to me.

Reply:
@brjohnstmsft this is not nececary, as mentioned this is a validation of data before handling it. This is not a bug, just tight null validtion :).